monitor/redis/watcher.go

```diff
@@ -70,20 +70,35 @@ func (w *Watcher) monitor(ctx context.Context, ch chan<- []*change.Change) {
 	}
 }
 
+func toPtr(s string) *string {
+	return &s
+}
+
 func (w *Watcher) getValues(ctx context.Context, ch chan<- []*change.Change) {
-	values, err := w.client.MGet(ctx, w.keys...).Result()
-	if err != nil {
-		log.Errorf("failed to MGET keys %v: %v", w.keys, err)
-		return
+	values := make([]*string, len(w.keys))
+	for i, key := range w.keys {
+		strCmd := w.client.Get(ctx, key)
+		if strCmd == nil {
+			log.Errorf("failed to get value for key %s: nil strCmd", key)
+			continue
+		}
+		// notice: if the key is not found we default to the same behavior as if
+		// the key was found with an empty value string
+		if strCmd.Err() != nil && strCmd.Err() != redis.Nil {
```

Comment:
we can now trigger update to `""` in case the key was deleted generating consistency across pods